webserver/views/datasets.py

```diff
@@ -89,6 +96,37 @@ def view(id):
     )
 
 
+@datasets_bp.route("/<uuid:dataset_id>/download_dataset")
+def download_dataset(dataset_id):
+    ds = get_dataset(dataset_id)
+    zip_file = create_low_level_files(ds)
+    # Download the zip file
+    return send_file(zip_file, as_attachment=True)
+
+
+def create_low_level_files(ds):
+    """Create a zip archive of all items in the dataset
+    with folders for each class and files in each folder
+    containing low-level json data
+    """
+    utils.path.create_path(zipped_dataset_dir)
+    dataset_dir = os.path.join(os.path.abspath(zipped_dataset_dir), ds["name"])
```

Comment:
After talking with @zas a bit we decided that using python-zipstream is probably a better choice for us than mod_zip at the moment (it means that we can use operating system updates for any possible security updates in nginx).
We will need to monitor how long it takes to generate these zip files, especially for large datasets, as this operation will block a flask backend while creating the zip file.
We can write log messages with the dataset id and start/end time of generating the zip file so that we can analyse it in the future.
If we find that it affects performance we might have to move back to a method where we generate the zip file and then serve it, but we should consider generating the file in the background using a process worker like celery.